### PR TITLE
OpenAPI - vo manager implemented

### PR DESCRIPTION
**Pull request by xkureck**
* implement openAPI methods for vosManager
* added new schemas and responses

### CODE REVIEW

File: perun-openapi/openapi.yml [hunk 1]
```diff
@@ -6832,11 +6868,104 @@ paths:
...
+  /urlinjsonout/vosManager/deleteVo/withAll:
+    post:
+      tags:
+        - VosManager
+      operationId: deleteVoForce
+      summary: Deletes a VO. If <code>force == true</code> then VO is deleted including members, groups and resources.
+               Otherwise only empty VO is deleted or exception is thrown.
+      parameters:
+        - $ref: '#/components/parameters/voId'
+        - { name: force, description: "Force must be true", schema: { type: boolean },  in: query, required: true }
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]: There is no need to have two separate methods deleteVo and deleteVoForce, the "force" parameter can be declared "required: false" which makes it optional. Keep just one deleteVo method with the optional parameter.

File: perun-openapi/openapi.yml [hunk 2]
```diff
@@ -6866,11 +6995,126 @@ paths:
...
+        - VosManager
+      operationId: findCandidatesWithMaximumNumberResults
+      summary: |
+              Find candidates for VO with specified maximum number of results. Candidates can be used to create new members.
+              Candidates are searched in VOs external sources (if available). Candidates, which are already members of VO are never
+              returned even if they match searchString.
+      parameters:
+        - $ref: '#/components/parameters/id'
+        - { name: searchString, schema: { type: string}, in: query, required: true, description: 'Text to search by' }
+        - { name: maxNumOfResults, schema: { type: integer}, in: query, required: true, description: 'Number of maximum results' }
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]: The operations /json/vosManager/findCandidates and /json/vosManager/findCandidates/withMaxNumberResults can be merged into one with optional parameter maxNumOfResults.

File: perun-openapi/openapi.yml [hunk 3]
```diff
@@ -6866,11 +6995,126 @@ paths:
...
+        - VosManager
+      operationId: getCompleteCandidatesForVo
+      summary: |
+                Find MemberCandidates for VO. MemberCandidates can be used to create new members. They are made of Candidate,
+                RichUser and Member objects. Candidates are searched in VO's external sources (if available). RichUsers are
+                searched in given VO and are associated with appropriate candidate and member. RichUsers for MemberCandidates
+                may also be searched in the whole Perun.
+      parameters:
+        - { name: vo, schema: { type: integer}, in: query, required: true, description: 'vo id' }
+        - name: attrNames
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]: Can be replaced with $ref: '#/components/parameters/attrNames'

File: perun-openapi/openapi.yml [hunk 4]
```diff
@@ -6866,11 +6995,126 @@ paths:
...
+        - VosManager
+      operationId: getCompleteCandidatesForGroup
+      summary: |
+                Find MemberCandidates for GROUP. MemberCandidates can be used to create new members. They are made of Candidate,
+                RichUser and Member objects. Candidates are searched in VO's or GROUP's (depends on privileges) external sources
+                (if available). RichUsers are searched in given VO and are associated with appropriate candidate and member.
+                RichUsers for appropriate Candidate are also searched in the whole Perun.
+      parameters:
+        - { name: group, schema: { type: integer}, in: query, required: true, description: 'group id' }
+        - name: attrNames
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]: dtto

File: perun-openapi/openapi.yml [hunk 5]
```diff
@@ -6934,7 +7206,35 @@ paths:
...
+
+  /urlinjsonout/vosManager/removeSponsorRole/group:
+    post:
+      tags:
+        - VosManager
+      operationId: removeSponsorRoleFromGroup
+      summary: Removes group as a sponsor. All group members will cease to be sponsors, and their sponsored members will be set as expired if the group member was their last sponsor.
+      parameters:
+        - $ref: '#/components/parameters/voId'
+        - { name: authorizedGroup, schema: { type: integer}, in: query, required: true, description: 'Group id' }
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]: The authorizedGroup parameter is repeated at methods addSponsorRoleToGroup, removeSponsorRoleFromGroup, addVoAdminGroup, removeVoAdminGroup, addFacilityAdminGroup, removeFacilityAdminGroup, addAdminToGroup, removeGroupAdminResource. Please declare it in #/components/parameters and use $ref to reference it.

File: perun-openapi/openapi.yml [hunk 6]
```diff
@@ -6972,8 +7272,34 @@ paths:
...
+      description: |
+        Get list of all richUser administrators for the vo and supported role with specific attributes.
+        Supported roles: VOOBSERVER, TOPGROUPCREATOR, VOADMIN
+        If "onlyDirectAdmins" is == true, return only direct admins of the vo for supported role with specific attributes.
+        If "allUserAttributes" is == true, do not specify attributes through list and return them all in objects richUser. Ignoring list of specific attributes.
+      parameters:
+        - $ref: '#/components/parameters/voId'
+        - { name: role, schema: { $ref: '#/components/schemas/VoAdminRoles' }, in: query, required: true, description: 'role name'}
+        - name: specificAttributes
+          description: 'list of specified attributes which are needed in object richUser'
```
Comment (martin-kuba) [inline perun-openapi/openapi.yml]: Please use the shorter "flow mapping" syntax as for the other parameters.

**Comment by Vojtech-Sassmann** [comment]
Hi @xkureck. Please, rebase this PR.

**Pull request merged**